search/binary_search_tree.py

```diff
@@ -11,8 +13,14 @@ def depth_first_for_each(self, cb):
         if self.right:
             self.right.depth_first_for_each(cb)
 
-    def breadth_first_for_each(self, cb):
-        pass
+    def breadth_first_for_each(self, cb, queue=deque()):
+        cb(self.value)
+        if self.left:
+            queue.append(self.left)
+        if self.right:
+            queue.append(self.right)
+        if len(queue) > 0:
+            queue.popleft().breadth_first_for_each(cb, queue)
```

Comment:
great this is very similar to depth first but popleft ensures it's a breadth first traversal.